datafusion/functions/src/datetime/to_local_time.rs

```diff
@@ -348,6 +327,25 @@ impl ScalarUDFImpl for ToLocalTimeFunc {
 
         self.to_local_time(args)
     }
+
+    fn coerce_types(&self, arg_types: &[DataType]) -> Result<Vec<DataType>> {
+        if arg_types.len() != 1 {
+            return Err(DataFusionError::Execution(format!(
+                "to_local_time function requires 1 argument, got {:?}",
+                arg_types.len()
+            )));
+        }
+
+        match &arg_types[0] {
+            Timestamp(Nanosecond, timezone) => Ok(vec![Timestamp(Nanosecond, timezone.clone())]),
+            Timestamp(Microsecond, timezone) => Ok(vec![Timestamp(Microsecond, timezone.clone())]),
+            Timestamp(Millisecond, timezone) => Ok(vec![Timestamp(Millisecond, timezone.clone())]),
+            Timestamp(Second, timezone) => Ok(vec![Timestamp(Second, timezone.clone())]),
+            _ => Err(DataFusionError::Execution(format!(
```

Comment:
You could use `macro` like `exec_err`, `plan_err`

Reply:
I tried to use the macros, but that caused the sqllogictest to fail. As, you can see in [here](https://github.com/mesejo/arrow-datafusion/actions/runs/10788101958/job/29918150107):

```
 External error: statement is expected to fail with error:
	(regex) DataFusion error: Error during planning: Execution error: User\-defined coercion failed with Plan\("The to_local_time function can only accept timestamp as the arg, got Utf8"\)
but got error:
	DataFusion error: Error during planning: Execution error: User-defined coercion failed with Plan("The to_local_time function can only accept timestamp as the arg, got Utf8
```
The messages match, but the error type, even though it has the same name, is different.

I could not figure out what was causing the macros to fail.

Reply:
You could try to rewrite the error with `The to_local_time function can only accept timestamp as the arg. got Utf8`.

Reply:
Thanks! Done using plan_err!